### ISSUE TITLE
Program under Docker takes too long to start

### ISSUE THREAD

**Issue by davidmartos96** [dart-lang/sdk#53550]
Hello, I've recently encountered with a situation where I needed access to the vm service inside a docker container. This is in order to evaluate Dart expressions on a containerized interpreter.

What I encountered is that a small program which uses the flag `--enable-vm-service` when running takes 3 whole minutes to start under Docker vs essentially instant on my local machine.
I'm not sure what is the difference. How could I make the Docker image behave so that it has better start time, like locally? 

Here is the program: https://github.com/davidmartos96/dart_vm_docker_demo

### Dart info
- Dart 3.1.1 (stable) (Tue Sep 5 12:20:14 2023 +0000) on "linux_x64"
- on linux / Linux 6.4.15-200.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Sep  7 00:25:01 UTC 2023
- locale is en_US.UTF-8

I paste some parts from the repo here for context:

### Steps to run locally

`dart run --enable-vm-service bin/main.dart`

### Steps to run in Docker

`docker build -t vm_demo .`

`docker run --rm vm_demo`

### Dockerfile

```dockerfile
FROM dart:3.1.1

WORKDIR /app
COPY pubspec.yaml .
RUN dart pub get

COPY . .
RUN dart pub get --offline
RUN dart compile kernel -o bin/main.dill bin/main.dart

ENTRYPOINT ["dart", "run", "--enable-vm-service", "/app/bin/main.dill"]
```

**Comment by a-siva** [comment]
@davidmartos96 when the `--enable-vm-service` option is not used does the program start and terminate instantly when run in docker ? Could you true `dart run --enable-vm-service --no-dds app/bin/main.dill` and see if that makes it better

**Comment by davidmartos96** [comment]
@a-siva Yes, it's only when using the vm flag when it takes too long to start.

The --no-dds flag made it run instantly though, thanks!

**Comment by a-siva** [comment]
> The --no-dds flag made it run instantly though, thanks!

When `--enable-vm-service` is specified we start an additional dart process called the dart debug server and I am wondering if you have provisioned docker for only one Dart process and this additional dart processes causes things to slow down

**Comment by davidmartos96** [comment]
@a-siva I haven't configured docker in any special way. Above is the hello world application with the Docker file that reproduces this.

It doesn't feel slow, but that it waits for something to be ready (the DDS). It always takes the same amount of time to start.

Are you able to reproduce it or is it device dependent based on some docker configuration?

**Comment by a-siva** [comment]
> Are you able to reproduce it or is it device dependent based on some docker configuration?

I haven't tried to reproduce it yet, let me do that and report back.
